### PR TITLE
ESEF 2023

### PR DESCRIPTION
**Pull request by RobinetDenisAcsone**
#### Reason for change
The esma published an updated version of the reporting manual for esef 2023

#### Description of change
Add the changed rules.

#### Steps to Test
Open an esef report and select the disclosure system 2023, no conformance suite was released (yet?)

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/ESEF/resources/config.xml [hunk 1]
```diff
@@ -5,7 +5,27 @@
...
+         description="ESMA RTS on ESEF-2023 Validation Checks for Unconsolidated Financial Statements"
+         validationType="ESEF"
+         exclusiveTypesPattern="EFM|GFM|FERC|HMRC|SBR.NL|EBA|EIOPA|ESEF"
+         blockDisallowedReferences="false"
+         validateFileText="false"
+         contextElement="scenario"
+         />
+
+    <DisclosureSystem
+            names="ESMA RTS on ESEF-2022|ESEF-2022|esef-2022"
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/resources/config.xml:28]: I think we should keep 2022 as the default (`esef` and `esef-unconsolidated` disclosure systems) until the conformance suite for 2023 is released.

File: arelle/plugin/validate/ESEF/ESEF_Current/DTS.py [hunk 2]
```diff
@@ -38,7 +39,7 @@ def checkFilingDTS(val: ValidateXbrl, modelDocument: ModelDocument, esefNotesCon
 
     isExtensionDoc = isExtension(val, modelDocument)
     filenamePattern = filenameRegex = None
-    anchorAbstractExtensionElements = val.authParam["extensionElementsAnchoring"] == "include abstract"
+    anchorAbstractExtensionElements = is2022DisclosureSystem(val.modelXbrl) and val.authParam["extensionElementsAnchoring"] == "include abstract"
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/ESEF_Current/DTS.py:42]: Here and elsewhere, this looks like it should be checking if the disclosure system is older than 2023 (2021 or 2022)?

File: arelle/plugin/validate/ESEF/ESEF_Current/DTS.py [hunk 3]
```diff
@@ -108,7 +109,7 @@ def lc3wordAdjust(word: str) -> str:
                         val.modelXbrl.error("ESEF.3.4.3.extensionTaxonomyDimensionNotAssignedDefaultMemberInDedicatedPlaceholder",
                             _("Each dimension in an issuer specific extension taxonomy MUST be assigned to a default member in the ELR with role URI http://www.esma.europa.eu/xbrl/role/core/ifrs-dim_role-990000 defined in esef_cor.xsd schema file. %(qname)s"),
                             modelObject=modelConcept, qname=modelConcept.qname)
-                    if modelConcept.isDomainMember and modelConcept in val.domainMembers and modelConcept.typeQname not in qnDomainItemTypes:
+                    if modelConcept.isDomainMember and modelConcept in val.domainMembers and ((is2022DisclosureSystem(val.modelXbrl) and modelConcept.typeQname not in qnDomainItemTypes) or (not is2022DisclosureSystem(val.modelXbrl) and modelConcept.typeQname not in qnDomainItemTypes2023)):
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/ESEF_Current/DTS.py:112]: Suggest moving the year disclosure system selection out of the if statement:
```suggestion
                    esefDomainItemTypes = qnDomainItemTypes if getDisclosureSystemYear(val.modelXbrl) < 2023 else qnDomainItemTypes2023
                    if modelConcept.isDomainMember and modelConcept in val.domainMembers and modelConcept.typeQname not in esefDomainItemTypes:
```

File: arelle/plugin/validate/ESEF/Util.py [hunk 4]
```diff
@@ -51,6 +51,10 @@ def isInEsefTaxonomy(val: ValidateXbrl, modelObject: ModelObject | None) -> bool
     return (any(ns.startswith(esefNsPrefix) for esefNsPrefix in esefTaxonomyNamespaceURIs))
 
 
+def is2022DisclosureSystem(modelXbrl: ModelXbrl) -> bool:
+    return any("2022" in name for name in modelXbrl.modelManager.disclosureSystem.names)
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/Util.py:55]: Suggest replacing this (and the preexisting `is2021DisclosureSystem` function) with a generic get disclosure system year function and using that to see if the year is newer or older than a specfic year.

```python
YEAR_GROUP = "year"
DISCLOSURE_SYSTEM_YEAR_PATTERN = re.compile(rf"esef-(?P<{YEAR_GROUP}>20\d\d)")


def getDisclosureSystemYear(modelXbrl: ModelXbrl) -> int:
    for name in modelXbrl.modelManager.disclosureSystem.names:
        disclosureSystemYear = DISCLOSURE_SYSTEM_YEAR_PATTERN.fullmatch(name)
        if disclosureSystemYear:
            return int(disclosureSystemYear.group(YEAR_GROUP))
    raise ValueError(f"Unable to determine year of ESEF disclosure system matching pattern 'esef-20XX' from {modelXbrl.modelManager.disclosureSystem.names}")

...
if getDisclosureSystemYear(modelXbrl) >= 2023:
    # 2023 logic
else:
    # 2021 and 2022 logic
```

**Comment by RobinetDenisAcsone** [comment]
@austinmatherne-wk Changes after review done, and rule 1.9.1 added

**Review by austinmatherne-wk — DISMISSED** [review]

File: arelle/plugin/validate/ESEF/Util.py [hunk 5]
```diff
@@ -27,6 +28,17 @@
 
 _: TypeGetText
 
+YEAR_GROUP = "year"
+DISCLOSURE_SYSTEM_YEAR_PATTERN = re.compile(rf"esef-(?P<{YEAR_GROUP}>20\d\d)")
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/Util.py:32]: Mistake on my part in my suggestion. This also needs to be able to parse the unconsolidated names.

```suggestion
DISCLOSURE_SYSTEM_YEAR_PATTERN = re.compile(rf"esef-(?:unconsolidated-)?(?P<{YEAR_GROUP}>20\d\d)")
```

Comment (RobinetDenisAcsone) [reply arelle/plugin/validate/ESEF/Util.py]: Changed, sorry, I didn't check with the unconsolidated, I tried it before pushing this time

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**